src/main/java/at/ac/tuwien/kr/alpha/solver/heuristics/BerkMin.java

```diff
@@ -56,25 +56,31 @@
 	public static final int DEFAULT_DECAY_AGE = 10;
 	public static final double DEFAULT_DECAY_FACTOR = 0.25;
 
+	private final Random rand;
+
 	private Assignment assignment;
 	private Map<Integer, Double> activityCounters = new HashMap<>();
 	private Map<Integer, Integer> signCounters = new HashMap<>();
 	private Deque<NoGood> stackOfNoGoods = new ArrayDeque<>();
 	private int decayAge;
 	private double decayFactor;
 	private int stepsSinceLastDecay;
-	private Random rand = new Random();
 	private Predicate<? super Integer> isAtomActiveChoicePoint;
 
-	public BerkMin(Assignment assignment, Predicate<? super Integer> isAtomActiveChoicePoint, int decayAge, double decayFactor) {
+	public BerkMin(Assignment assignment, Predicate<? super Integer> isAtomActiveChoicePoint, int decayAge, double decayFactor, Random random) {
 		this.assignment = assignment;
 		this.isAtomActiveChoicePoint = isAtomActiveChoicePoint;
 		this.decayAge = decayAge;
 		this.decayFactor = decayFactor;
+		this.rand = random;
+	}
+
+	public BerkMin(Assignment assignment, Predicate<? super Integer> isAtomActiveChoicePoint, Random random) {
+		this(assignment, isAtomActiveChoicePoint, DEFAULT_DECAY_AGE, DEFAULT_DECAY_FACTOR, random);
 	}
 
 	public BerkMin(Assignment assignment, Predicate<? super Integer> isAtomActiveChoicePoint) {
-		this(assignment, isAtomActiveChoicePoint, DEFAULT_DECAY_AGE, DEFAULT_DECAY_FACTOR);
+		this(assignment, isAtomActiveChoicePoint, DEFAULT_DECAY_AGE, DEFAULT_DECAY_FACTOR, new Random());
```

Comment:
I wanted to keep the "all default" constructor, that will construct it's own `Random`. This will be used for example in `BerkMinTest`. This way the constructor stays compatible with existing code. I agree that always requiring some source of randomness to be passed would be more strict. However you like.